### PR TITLE
Added filter and converter infrastructure.

### PR DESCRIPTION
**Pull request by allotria**
I added a simple infrastructure to plug-in filters and converters.

The filters are inspired by your earlier couchbase-kafka-connector.

### CODE REVIEW

**Comment by avsej** [comment]
Hi @allotria, thanks for the pull request!! 

To get this change in and collaborate in code review, please register on Gerrit and accept our CLA. The easiest way to do this is to follow the link below, sign in with your GitHub account and then follow through the steps provided on that page to sign an 'Individual' agreement: http://review.couchbase.org/#/settings/new-agreement.

Note: Please contact us if you have any issues registering with Gerrit!

File: src/main/java/com/couchbase/connect/kafka/CouchbaseSourceTask.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
 /**
  * Copyright 2016 Couchbase, Inc.
- *
+ * <p>
```
Comment (avsej) [inline src/main/java/com/couchbase/connect/kafka/CouchbaseSourceTask.java]: could you configure your editor not to add `<p>` tags in javadoc?

File: src/main/java/com/couchbase/connect/kafka/CouchbaseSourceTask.java [hunk 2]
```diff
@@ -43,9 +31,15 @@
...
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class CouchbaseSourceTask extends SourceTask {
-    private static final Logger LOGGER = LoggerFactory.getLogger(CouchbaseSourceConnector.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(CouchbaseSourceTask.class);
```
Comment (avsej) [inline src/main/java/com/couchbase/connect/kafka/CouchbaseSourceTask.java]: I've fixed it separately on master

File: src/main/java/com/couchbase/connect/kafka/CouchbaseSourceTask.java [hunk 3]
```diff
@@ -82,10 +74,15 @@ public void start(Map<String, String> properties) {
...
+        filter = createFilter(config.getString(CouchbaseSourceConnectorConfig.EVENT_FILTER_CLASS_CONFIG));
+        converter = createConverter(
+                config.getString(CouchbaseSourceConnectorConfig.DCP_MESSAGE_CONVERTER_CLASS_CONFIG));
+
         topic = config.getString(CouchbaseSourceConnectorConfig.TOPIC_NAME_CONFIG);
         bucket = config.getString(CouchbaseSourceConnectorConfig.CONNECTION_BUCKET_CONFIG);
         String password = config.getString(CouchbaseSourceConnectorConfig.CONNECTION_PASSWORD_CONFIG);
-        List<String> clusterAddress = config.getListWorkaround(CouchbaseSourceConnectorConfig.CONNECTION_CLUSTER_ADDRESS_CONFIG);
+        List<String> clusterAddress = config
+                .getListWorkaround(CouchbaseSourceConnectorConfig.CONNECTION_CLUSTER_ADDRESS_CONFIG);
```
Comment (avsej) [inline src/main/java/com/couchbase/connect/kafka/CouchbaseSourceTask.java]: could you preserve previous formatting?

File: src/main/java/com/couchbase/connect/kafka/CouchbaseSourceTask.java [hunk 4]
```diff
@@ -124,19 +121,51 @@ public void start(Map<String, String> properties) {
...
+        if (className != null && !"".equals(className)) {
+            try {
+                return (Filter) Class.forName(className).newInstance();
+            } catch (InstantiationException e) {
+                throw new ConnectException("Couldn't create filter in CouchbaseSourceTask due to an error", e);
+            } catch (IllegalAccessException e) {
+                throw new ConnectException("Couldn't create filter in CouchbaseSourceTask due to an error", e);
+            } catch (ClassNotFoundException e) {
+                throw new ConnectException("Couldn't create filter in CouchbaseSourceTask due to an error", e);
+            }
```
Comment (avsej) [inline src/main/java/com/couchbase/connect/kafka/CouchbaseSourceTask.java]: could you extract this method as utility or try to use `org.apache.kafka.common.utils.Utils.newInstance()`

File: src/main/java/com/couchbase/connect/kafka/converter/Converter.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.couchbase.connect.kafka.converter;
```
Comment (avsej) [inline src/main/java/com/couchbase/connect/kafka/converter/Converter.java]: Make sure that all new files contain our license header

**Comment by allotria** [comment]
I accepted the CLA in Gerrit. Do I need to print, sign, scan and mail it to you or does it count when I wrote "I ACCEPT" into the form?

**Comment by allotria** [comment]
I fixed your code review change requests.

**Comment by avsej** [comment]
Thank you @allotria

**Pull request merged**